src/java.base/share/classes/java/util/Pair.java

```diff
@@ -0,0 +1,162 @@
+     *
+     * @param first the first element of the pair
+     * @param second the second element of the pair
+     */
+    public Pair(K first, V second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    /**
+     * Returns the first element of the pair.
+     *
+     * @return the first element of the pair
+     */
+    public K getFirst() {
+        return first;
+    }
+
+    /**
+     * Sets the first element of the pair.
+     *
+     * @param first the first element of the pair
+     */
+    public void setFirst(K first) {
+        this.first = first;
+    }
+
+    /**
+     * Returns the second element of the pair.
+     *
+     * @return the second element of the pair
+     */
+    public V getSecond() {
+        return second;
+    }
+
+    /**
+     * Sets the second element of the pair.
+     *
+     * @param second the second element of the pair
+     */
+    public void setSecond(V second) {
+        this.second = second;
+    }
+
+    /**
+     * Compares this pair with the specified pair for order.
+     * <p>
+     * Comparison is based first on the {@code first} elements, and if they are equal,
+     * then on the {@code second} elements. Both elements are compared using their natural ordering,
+     * unless they are {@code null}, in which case {@code null} is considered less than non-null.
+     * <p>
+     * This method is {@code null}-safe and consistent with equals when both elements are non-null.
+     *
+     * @param other the pair to be compared
+     * @return a negative integer, zero, or a positive integer as this pair
+     *         is less than, equal to, or greater than the specified pair
+     * @throws NullPointerException if {@code other} is {@code null}
+     */
+    @Override
+    public int compareTo(Pair<K, V> other) {
+        if (other == null) {
+            throw new NullPointerException("Compared pair must not be null");
+        }
+
+        int cmp = compareNullable(this.first, other.first);
+        if (cmp != 0) {
+            return cmp;
+        }
+
+        return compareNullable(this.second, other.second);
+    }
+
+    /**
+     * Helper method to compare two Comparable objects, handling nulls.
+     * {@code null} is considered less than any non-null value.
+     *
+     * @param a first value
+     * @param b second value
+     * @return comparison result
+     */
+    @SuppressWarnings("unchecked")
+    private <T extends Comparable<? super T>> int compareNullable(T a, T b) {
+        if (a == b) return 0;
+        if (a == null) return -1;
+        if (b == null) return 1;
+        return a.compareTo(b);
+    }
+
+    /**
+     * Returns a string representation of this pair in the format {@code (first, second)}.
+     *
+     * @return a string representation of this pair
+     */
+    @Override
+    public String toString() {
+        return "(" + String.valueOf(first) + ", " + String.valueOf(second) + ")";

```

Comment:
```suggestion
        return "(" + first + ", " + second + ")";
```
StringConcation will handle null, no need to call String.valueOf here

Reply:
THANKS for pointing out, will do!!

Reply:
Hey @wenshao 
It would be so mnice of you, if you could review https://github.com/openjdk/jdk/pull/25059